### DISCUSSION TITLE
Using `or` in a `select`

### DISCUSSION THREAD

**Discussion by joelanford** [mikefarah/yq#804] [Q&A] [answered]
I have a multi-document YAML file and I'm trying to filter it down to a smaller set of documents based on a boolean expression that uses `or`. Here's a reproducer:

### Version
yq version: 4.7.1

### Input
File `test.yaml`:
```yaml
---
a: hello
---
b: world
```

`yq` command:
```bash
yq e 'select(.a == "hello" or .b == "world")' test.yaml
```
### Expected Output
```yaml
a: hello
---
b: world
```

### Actual Output
Nothing is printed
```yaml
```

### Background
I tried removing the select to see if I somehow messed up my boolean expression:
`yq` command:
```bash
yq e '.a == "hello" or .b == "world"' test.yaml
```

And the output here is what I would have expected:
```yaml
true
---
true
```

Am I doing something wrong or have I discovered a bug?

**Comment by mikefarah** [comment, marked as answer]
quick look at this and it looks like it's a bug. Can't work out what's going on - need to investigate further.

As a work around this works:

```
select(.a == "hello"), select(.b == "world")
```

**Reply by mikefarah** [reply]
@joelanford this has been fixed in 4.8.0 :)
